src/ecs/resources/input.rs

```diff
@@ -62,45 +92,63 @@ impl<'a> Iterator for PressedMouseButtons<'a> {
 /// ```
 #[derive(Default)]
 pub struct InputHandler {
-    pressed_keys: HashMap<VirtualKeyCode, KeyQueryState>,
-    pressed_mouse_buttons: HashMap<MouseButton, KeyQueryState>,
+    pressed_keys: Vec<VirtualKeyCode>,
+    previous_pressed_keys: Vec<VirtualKeyCode>,
+    pressed_mouse_buttons: Vec<MouseButton>,
+    previous_pressed_mouse_buttons: Vec<MouseButton>,
     // defined to match glium
     mouse_position: Option<(i32, i32)>,
     previous_mouse_position: Option<(i32, i32)>,
+    axes: HashMap<i32, Axis>,
```

Comment:
On second thought it'd probably be better for this to be transparent right now, as the facilities for making input rebinding part of the asset system don't exist yet.  I'll probably see if I can do this again when #244 is merged.